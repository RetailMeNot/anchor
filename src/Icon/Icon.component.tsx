// REACT
import * as React from 'react';
// VENDOR
import classNames from 'classnames';
import styled from 'styled-components';
// COMPONENTS
import {
  Hamburger,
  ChevronDown,
  ChevronUp,
  ChevronRight,
  Cart,
  Search
} from './SVGs';
// THEME

type IconTypes = 'hamburger' | 'chevron-down' | 'chevron-up' | 'chevron-right' | 'cart' | 'search';

export interface IconSVGProps {
  width?: number;
  height?: number;
  color?: string;
}

interface IconProps {
  type: IconTypes;
  children?: any;
  className?: string;
  color?: string;
  width?: number;
  height?: number;
}

export const StyledIcon = styled.div`
  display: inline-flex;
  max-width: 1.5rem;
  justify-content: end;

  > svg {
    flex: 0 0 1.25rem;
  }
`;

const DefaultProps: IconProps = {
  type: 'hamburger', // TODO: default icon
  color: '#222',
};

const Icons = {
  'hamburger': Hamburger,
  'chevron-down': ChevronDown,
  'chevron-up': ChevronUp,
  'chevron-right': ChevronRight,
  'cart': Cart,
  'search': Search,
};

export const Icon = ({
  className,
  type,
  color,
  width,
  height,
  ...props
}: IconProps = DefaultProps): React.ReactElement<any> => (
  <StyledIcon className={classNames(className)} {...props}>
    {Icons[type]({ color, width, height })}
  </StyledIcon>
);

Icon.defaultProps = DefaultProps;

export default Icon;
