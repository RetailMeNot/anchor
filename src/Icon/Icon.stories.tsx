// REACT
import * as React from 'react';
// STORYBOOK
import { storiesOf } from '@storybook/react';
// VENDOR
import styled from 'styled-components';
// COMPONENT
import Icon from './Icon.component';

const StyledStory = styled.div`
  padding: 2rem 5rem;
`;

storiesOf('Components/Icon', module)
  .add('All', () => (
      <StyledStory>
        <Icon type="hamburger" />
      </StyledStory>
  ));
