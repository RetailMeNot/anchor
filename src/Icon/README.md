# Icon

The Icon component is actually an object which holds all Icon components. 
Trying to render `<Icon />` will result in an error, but for example, `<Icon.ArrowBack />` would not.

## Usage
    import { Icon } from '@retailmenot/anchor';
    
    const example = props => (
        <Icon.ArrowBack />
    );
    
Alternatively, each icon can be imported individually.
    
    import { IconArrowBack } from '@retailmenot/anchor';
    
    const example = props => (
        <IconArrowBack />
    );    
    
## Settings/Props

The component can accept 3 `props`:

* `color`: Defaults to **`#222`** but if passed it will take whatever hex/rgb/rgba value provided.
* `scale`: Defaults to **`md`**. The full range of options are **`'xs', 'sm', 'md', 'lg', 'xl', 'xxl'`**
* `className`: An additional class name. All icons have the **`anchor-icon`** class name in addition to a more specific class.
Ex: **`class="anchor-icon arrow-back"`**.    
        
    

