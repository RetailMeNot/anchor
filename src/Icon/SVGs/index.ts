export { default as Hamburger } from './Hamburger.svg';

export { default as ChevronDown } from './ChevronDown.svg';
export { default as ChevronUp } from './ChevronUp.svg';
export { default as ChevronRight } from './ChevronRight.svg';

export { default as Cart } from './Cart';
export { default as Search } from './Search';
