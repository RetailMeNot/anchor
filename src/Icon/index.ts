export { Icon } from './Icon.component';
