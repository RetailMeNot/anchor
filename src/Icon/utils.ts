// VENDOR
import styled from '@xstyled/styled-components';
import { margin as marginStyles } from '../theme/system.theme';

type ScaleFactors = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl';

export interface IconSVGProps {
    scale?: ScaleFactors;
    color?: string;
    className?: string;
    children?: any;
}

export const Scale = {
    xs: 8,
    sm: 14,
    md: 16,
    lg: 24,
    xl: 32,
    xxl: 48,
};

export const DefaultColor = 'currentColor';
export const DefaultScale = 'md';

export const StyledIcon = styled.span<IconSVGProps>`
    ${marginStyles}
    display: inline-block;
    height: ${({ scale = 'md' }: IconSVGProps) => `${Scale[scale] / 16}rem`};
    width: ${({ scale = 'md' }: IconSVGProps) => `${Scale[scale] / 16}rem`};
    line-height: 0;
`;

StyledIcon.displayName = 'AnchorIcon';

export type AnchorIcons =
    | 'AddEvent'
    | 'ArrowBack'
    | 'ArrowForward'
    | 'AvatarIcon'
    | 'AvatarOutline'
    | 'BarCode'
    | 'BulletList'
    | 'Calendar'
    | 'Camera'
    | 'Cart'
    | 'CashBack'
    | 'Cells'
    | 'Chat'
    | 'Check'
    | 'CheckSmall'
    | 'ChevronDown'
    | 'ChevronDownSmall'
    | 'ChevronLeft'
    | 'ChevronLeftSmall'
    | 'ChevronRight'
    | 'ChevronRightSmall'
    | 'ChevronUp'
    | 'ChevronUpSmall'
    | 'Clock'
    | 'Close'
    | 'CloseSmall'
    | 'CommentMore'
    | 'CreditCard'
    | 'CrossHairs'
    | 'Cut'
    | 'Disabled'
    | 'Dislike'
    | 'Download'
    | 'Ellipses'
    | 'EllipsesVertical'
    | 'Envelope'
    | 'EnvelopeOpen'
    | 'Error'
    | 'Expand'
    | 'Gear'
    | 'GiftCard'
    | 'Hamburger'
    | 'Heart'
    | 'HeartOutline'
    | 'Home'
    | 'Info'
    | 'InfoOutline'
    | 'Laptop'
    | 'Lightning'
    | 'Like'
    | 'ListIcon'
    | 'Lock'
    | 'Map'
    | 'Marker'
    | 'MarkerOutline'
    | 'Mobile'
    | 'News'
    | 'Pencil'
    | 'Play'
    | 'Plus'
    | 'PlusSmall'
    | 'Print'
    | 'Question'
    | 'QuestionOutline'
    | 'Refresh'
    | 'RetailMeNotLogo'
    | 'SadFace'
    | 'Search'
    | 'Share'
    | 'Sliders'
    | 'Star'
    | 'StarHalf'
    | 'StarOutline'
    | 'Success'
    | 'SuccessOutline'
    | 'Tag'
    | 'TagAdd'
    | 'Upload';
