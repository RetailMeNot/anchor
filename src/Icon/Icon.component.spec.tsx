// REACT
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// COMPONENT
import Icon from './Icon.component';
// ENZYME
import { shallow, mount } from 'enzyme';

// TEST SETUP
const subject = <Icon type="hamburger"/>;
const wrapper = mount(subject);
const component = shallow(subject);

describe('Component: Button', () => {
    it('should be defined', () => {
        /* tslint:disable */
        console.log(subject);
        console.log(wrapper.debug());
        console.log(component.debug());

        /* tslint:enable */
        expect(subject).toBeDefined();
        expect(wrapper).toBeDefined();
        expect(component).toBeDefined();
        const tree = renderer.create(subject).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
