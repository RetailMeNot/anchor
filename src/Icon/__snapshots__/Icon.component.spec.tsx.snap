// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Button should match its snapshot 1`] = `
<div
  className="sc-bdVaJa hVMrQO"
>
  <svg
    height="20px"
    version="1.1"
    viewBox="0 0 26 20"
    width="26px"
  >
    <g
      fill="none"
      fillRule="evenodd"
      id="Page-1"
      stroke="none"
      strokeWidth="1"
    >
      <path
        d="M1.4625,4 L24.5375,4 C25.345125,4 26,3.105 26,2 C26,0.895 25.345125,0 24.5375,0 L1.4625,0 C0.654875,0 0,0.895 0,2 C0,3.105 0.654875,4 1.4625,4"
        fill="#222"
        id="Top-Fill"
      />
      <path
        d="M1.4625,12 L24.5375,12 C25.345125,12 26,11.105 26,10 C26,8.895 25.345125,8 24.5375,8 L1.4625,8 C0.654875,8 0,8.895 0,10 C0,11.105 0.654875,12 1.4625,12"
        fill="#222"
        id="Center-Fill"
      />
      <path
        d="M1.4625,20 L24.5375,20 C25.345125,20 26,19.105 26,18 C26,16.895 25.345125,16 24.5375,16 L1.4625,16 C0.654875,16 0,16.895 0,18 C0,19.105 0.654875,20 1.4625,20"
        fill="#222"
        id="Bottom-Fill"
      />
    </g>
  </svg>
</div>
`;
